Assets/Scritps/ArtifactSpawnManager.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.Experimental;
+using UnityEditor.Search;
+using UnityEngine;
+
+[Serializable]
+public struct SpecialArtifactSpawn
+{
+    public ArtifactController artifact;
+    public List<Transform> spawnPoints;
+
+}
+
+public class ArtifactSpawnManager : MonoBehaviour
+{
+    [Header("Special Artifacts")]
+    [Tooltip("The script will spawn the special items first.")]
+    [SerializeField]
+    private List<SpecialArtifactSpawn> specialArtifacts;
+
+    [Header("Simple Artifacts")]
+    [SerializeField]
+    private List<ArtifactController> simpleArtifacts;
+    [SerializeField]
+    private List<Transform> simpleSpawnPoints;
+
+    private void Awake()
+    {
+#if UNITY_EDITOR
+        CheckData();
+#endif
+
+        SpawnArtifacts();
+    }
+
+    private void SpawnArtifacts()
+    {
+        var usedSpawnPoints = new HashSet<Transform>();
+        foreach (var specialArtifact in specialArtifacts)
+            SpawnArtifact(specialArtifact.artifact, specialArtifact.spawnPoints, usedSpawnPoints);
+
+        simpleSpawnPoints.Shuffle(simpleSpawnPoints.Count * 2);
+        SpawnArtifact(new Queue<ArtifactController>(simpleArtifacts), new Queue<Transform>(simpleSpawnPoints), usedSpawnPoints);
+    }
+
+    private void CheckData()
```

Comment:
If you don't use this in any other function, can you put it also in a #if Unity_EDITOR preprocessor?

Reply:
I did it.